lib/logstash/filters/bytes.rb

```diff
@@ -0,0 +1,151 @@
+# encoding: utf-8
+require "logstash/filters/base"
+require "logstash/namespace"
+
+# This filter will parse a given string as a computer storage
+# value (e.g. "123 MB" or "5.3GB") and add a new numeric field with
+# size in bytes
+#
+class LogStash::Filters::Bytes < LogStash::Filters::Base
+
+  PREFIX_POWERS = {
+    'k' => 1, # 1 kilobyte = 1024 ^ 1 bytes
+    'm' => 2, # 1 megabyte = 1024 ^ 2 bytes
+    'g' => 3, # 1 gigabyte = 1024 ^ 3 bytes
+    't' => 4, # 1 terabyte = 1024 ^ 4 bytes
+    'p' => 5, # 1 petabyte = 1024 ^ 5 bytes
+  }.freeze
+
+  VALID_CONVERSION_METHODS = [ "binary", "metric" ]
+
+  DIGIT_GROUP_SEPARATORS = " _,."
+
+  # Setting the config_name here is required. This is how you
+  # configure this filter from your Logstash config.
+  #
+  # filter {
+  #   bytes {
+  #     source => "123 MB"
+  #     target => "num_bytes"
+  #   }
+  # }
+  #
+  config_name "bytes"
+  
+  # Source string, e.g. "123 MB", "5.3GB", etc.
+  config :source, :validate => :string, :default => "message"
+
+  # Target field name
+  config :target, :validate => :string
+
+  # Conversion method, either "binary" (1K = 1024B) or "metric" (1K = 1000B)
+  config :conversion_method, :validate => :string, :default => "binary"
```

Comment:
you can add some validation to this like here: https://github.com/logstash-plugins/logstash-input-tcp/blob/master/lib/logstash/inputs/tcp.rb#L80

Reply:
Sweet!